src/main/java/io/aiven/elasticsearch/repositories/gcs/GcsSettingsProvider.java

```diff
@@ -36,11 +37,17 @@
 
     private volatile Tuple<Storage, EncryptionKeyProvider> cachedSettings;
 
-    public Storage gcsClient() {
+    public Storage gcsClient() throws IOException {
+        if (Objects.isNull(cachedSettings) || Objects.isNull(cachedSettings.x())) {
+            throw new IOException("GCS client hasn't been configured");
```

Comment:
Looks more like `IllegalStateException` (same below)

Reply:
Must be `IOException` (I don't like it), otherwise ES swallows them, and you will see real reason in REST call `/_snapshot/status`

Reply:
🤦 but got it

Reply:
Maybe leave a comment there for posterity :)

Reply:
Ok I will. And play a bit with other options later ...

Reply:
added comment so far